src/main/java/seedu/address/logic/parser/CountryFilterCommandParser.java

```diff
@@ -0,0 +1,29 @@
+package seedu.address.logic.parser;
+
+import static seedu.address.logic.parser.CliSyntax.PREFIX_COUNTRY;
+
+import seedu.address.logic.commands.CountryFilterCommand;
+import seedu.address.logic.parser.exceptions.ParseException;
+import seedu.address.model.client.CountryMatchesInputCountryPredicate;
+import seedu.address.model.country.Country;
+import seedu.address.model.country.CountryManager;
+
+public class CountryFilterCommandParser implements Parser<CountryFilterCommand> {
+
+    @Override
+    public CountryFilterCommand parse(String args) throws ParseException {
+        ArgumentMultimap argMultimap =
+                ArgumentTokenizer.tokenize(args, PREFIX_COUNTRY);
+
+        if (argMultimap.getValue(PREFIX_COUNTRY).isEmpty()) {
+            throw new ParseException("invalid message"); // TODO: better messages
+        }
+        String countryCode = argMultimap.getValue(PREFIX_COUNTRY).get();
+        if (!CountryManager.isValidCode(countryCode)) {
+            throw new ParseException("invalid message"); // TODO: better messages
+        }
```

Comment:
This check seems to be carried out in `ParserUtil` already

Reply:
Ok I have removed the check